src/mlpack/bindings/cli/add_to_po.hpp

```diff
@@ -2,110 +2,101 @@
  * @file add_to_po.hpp
  * @author Ryan Curtin
  *
- * Utility functions to add options to boost::program_options based on their
- * type.
+ * Utility functions to add options to CLI11 based on their type.
  *
  * mlpack is free software; you may redistribute it and/or modify it under the
  * terms of the 3-clause BSD license.  You should have received a copy of the
  * 3-clause BSD license along with mlpack.  If not, see
  * http://www.opensource.org/licenses/BSD-3-Clause for more information.
  */
-#ifndef MLPACK_BINDINGS_CLI_ADD_TO_PO_HPP
-#define MLPACK_BINDINGS_CLI_ADD_TO_PO_HPP
+#ifndef MLPACK_BINDINGS_IO_ADD_TO_PO_HPP
+#define MLPACK_BINDINGS_IO_ADD_TO_PO_HPP
 
 #include <mlpack/core/util/param_data.hpp>
-#include <boost/program_options.hpp>
 #include <mlpack/core/util/is_std_vector.hpp>
 #include "map_parameter_name.hpp"
 
+#include <mlpack/third_party/CLI/CLI11.hpp>
+
 namespace mlpack {
 namespace bindings {
-namespace cli {
+namespace cmd {
 
 /**
- * Add a non-vector option to boost::program_options.
- *
- * @param boostName The name of the option to add to boost::program_options.
- * @param descr Description string for parameter.
- * @param desc Options description to add parameter to.
- */
-template<typename T>
-void AddToPO(const std::string& boostName,
-             const std::string& descr,
-             boost::program_options::options_description& desc,
-             const typename boost::disable_if<util::IsStdVector<T>>::type* = 0,
-             const typename boost::disable_if<std::is_same<T, bool>>::type* = 0)
-{
-  desc.add_options()(boostName.c_str(), boost::program_options::value<T>(),
-      descr.c_str());
-}
-
-/**
- * Add a vector option to boost::program_options.  This overload will use the
- * multitoken() option.
- *
- * @param boostName The name of the option to add to boost::program_options.
+ * Add a vector option to CLI11.
+ * 
+ * @param cliName The name of the option to add to CLI11.
  * @param descr Description string for parameter.
  * @param desc Options description to add parameter to.
  */
 template<typename T>
-void AddToPO(const std::string& boostName,
-             const std::string& descr,
-             boost::program_options::options_description& desc,
+void AddToPO(const std::string& cliName,
+             util::ParamData& param,
+             CLI::App& app,
              const typename boost::enable_if<util::IsStdVector<T>>::type* = 0,
              const typename boost::disable_if<std::is_same<T, bool>>::type* = 0)
 {
-  desc.add_options()(boostName.c_str(),
-      boost::program_options::value<T>()->multitoken(), descr.c_str());
+  app.add_option_function<T>(cliName.c_str(), 
+  [&param](const T& value)
+  {
+     param.value = value;
+     param.wasPassed = true;
+  }, 
+  param.desc.c_str());
```

Comment:
```suggestion
      [&param](const T& value)
      {
       param.value = value;
       param.wasPassed = true;
      }, 
      param.desc.c_str());
```
Thanks for restyling this; these are wrapped lines, though, so we should indent them twice.  Sorry to be so pedantic but I think it's important that we match style throughout the repository.  See also https://github.com/mlpack/mlpack/wiki/DesignGuidelines#style-guidelines

Reply:
I am very sorry for this but I did not find anything related to lamda in the StyleGuidline,